AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -1117,23 +1115,7 @@ - (ASScrollDirection)_scrollDirectionForVelocity:(CGPoint)scrollVelocity
 
 - (ASScrollDirection)scrollableDirections
 {
-  //Cache results of layoutInspector to ensure flags are up to date if getter lazily loads a new one.
-  id<ASCollectionViewLayoutInspecting> layoutInspector = self.layoutInspector;
-  if (_layoutInspectorFlags.scrollableDirections) {
-    return [layoutInspector scrollableDirections];
-  } else {
-    ASScrollDirection scrollableDirection = ASScrollDirectionNone;
-    CGFloat totalContentWidth = self.contentSize.width + self.contentInset.left + self.contentInset.right;
-    CGFloat totalContentHeight = self.contentSize.height + self.contentInset.top + self.contentInset.bottom;
-    
-    if (self.alwaysBounceHorizontal || totalContentWidth > self.bounds.size.width) { // Can scroll horizontally.
-      scrollableDirection |= ASScrollDirectionHorizontalDirections;
-    }
-    if (self.alwaysBounceVertical || totalContentHeight > self.bounds.size.height) { // Can scroll vertically.
-      scrollableDirection |= ASScrollDirectionVerticalDirections;
-    }
-    return scrollableDirection;
-  }
+  return [self.layoutInspector scrollableDirections];
```

Comment:
Might be good to add an AssertNotNil for the self.layoutInspector to ensure this object is always set.  I think the current code does ensure it's always set, but the assertion might help catch any future changes there.